net/netx/netx.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package netx contains the Network type.
+package netx
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/netip"
+	"sync"
+
+	"tailscale.com/net/memnet"
+)
+
+// Network is an interface for use in tests that describes either [RealNetwork]
```

Comment:
It's not just for use in tests now, I guess?

Could be done separately, but if this is going to become a non-test package it'd be nice to expend the package docs too, to give a bit more context.

Reply:
whoops, will fix.